src/App.js

```diff
@@ -4,25 +4,37 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 
-const useSemiPersistentState = () =>{
+function App() {
 
-    const [todoList, setTodoList] = React.useState(
-        JSON.parse(localStorage.getItem('savedTodoList') || '[]' )
-    );
+    const [todoList, setTodoList] = React.useState([]);
+
+    const [isLoading, setIsLoading] = React.useState(true);
 
 
     React.useEffect(() => {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList]);
 
-    return [todoList, setTodoList];
-}
+        new Promise((resolve, reject) =>
+            setTimeout(
+                () => resolve({data: {todoList:
+                            JSON.parse(localStorage.getItem('savedTodoList'))
+                }}),
+                2000
+            ),
+        ).then((result) => {
+            setTodoList(JSON.parse(localStorage.getItem('savedTodoList')))
```

Comment:
You should use `result` here. When you call `resolve` on line 18 what that will do is pass the arguments that you give `resolve` as the `result` here. Try it out. Set a breakpoint here and see what the contents of `result` are. Or try `console.log(result)`